test/extended/storage/mutable_csinode_allocatable.go

```diff
@@ -0,0 +1,678 @@
+		eniID := createAndAttachENI(ec2Client, instanceID)
+		testENIs[eniID] = true
+		e2e.Logf("Attached ENI: %q", eniID)
+
+		g.By("# Checking CSINode allocatable count should not update")
+		o.Consistently(func() int32 {
+			return getCSINodeAllocatableCountByDriver(ctx, oc, targetWorkerNode, ebsCSIDriverName)
+		}, 6*csiNodeUpdatePollInterval, csiNodeUpdatePollInterval).Should(o.Equal(originalAllocatableCount),
+			"CSINode allocatable count should not update less than nodeAllocatableUpdatePeriodSeconds")
+
+		g.By("# Checking CSINode allocatable count should not update")
+		storageClassName := GetCSIStorageClassByProvisioner(ctx, oc, ebsCSIDriverName)
+		e2e.Logf("Using StorageClass: %s", storageClassName)
+
+		// The StatefulSet will have replicas equal to the original allocatable count
+		// This should trigger ResourceExhausted errors for the last pod
+		replicas := originalAllocatableCount - targetWorkerNodeAttachedVolumeCount
+		statefulSetName := "storage-mca-test-sts"
+
+		g.By(fmt.Sprintf("# Creating StatefulSet %s with %d replicas scheduled to node %s", statefulSetName, replicas, targetWorkerNode))
+		statefulSet := createStatefulSetWithVolumeTemplate(oc.Namespace(), statefulSetName, replicas, targetWorkerNode, storageClassName)
+		_, err := oc.AdminKubeClient().AppsV1().StatefulSets(oc.Namespace()).Create(ctx, statefulSet, metav1.CreateOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred(), "failed to create StatefulSet")
+		g.DeferCleanup(cleanupStatefulSetWithPVCs, ctx, oc, oc.Namespace(), statefulSetName, replicas)
+
+		// Observe that the last replica pod transitions from ContainerCreating to Pending
+		// and that the ResourceExhausted error triggers an immediate CSINode allocatable count update
+		lastPodName := fmt.Sprintf("%s-%d", statefulSetName, replicas-1)
+		g.By(fmt.Sprintf("# Monitoring pod %s for ResourceExhausted error and CSINode update", lastPodName))
+		// Wait for the last pod to show ResourceExhausted error
+		o.Eventually(func() bool {
+			pod, err := oc.AdminKubeClient().CoreV1().Pods(oc.Namespace()).Get(ctx, lastPodName, metav1.GetOptions{})
+			if err != nil {
+				e2e.Logf("Pod %s not found yet: %v", lastPodName, err)
+				return false
+			}
+
+			// Check pod events for ResourceExhausted error
+			events, err := oc.AdminKubeClient().CoreV1().Events(oc.Namespace()).List(ctx, metav1.ListOptions{
+				FieldSelector: fmt.Sprintf("involvedObject.name=%s", lastPodName),
+			})
+			if err == nil {
+				for _, event := range events.Items {
+					if strings.Contains(event.Message, "ResourceExhausted") ||
+						strings.Contains(event.Message, "volume attach limit exceeded") ||
+						strings.Contains(event.Reason, "FailedAttachVolume") {
+						e2e.Logf("ResourceExhausted error detected for pod %s: %s - %s", lastPodName, event.Reason, event.Message)
+						return true
+					}
+				}
+			}
```

Comment:
There is `WaitTimeoutForEvent` [here](https://github.com/openshift/origin/blob/11e7d0abb839578a7ea3492e7851c9d9d854e01b/vendor/k8s.io/kubernetes/test/e2e/framework/events/events.go#L35), can it be used instead?

Reply:
Good to know, thanks! Updatded.